app/controllers/api/v1/owners_controller.rb

```diff
@@ -14,7 +14,7 @@ def show
   def create
     owner = User.find_by_name(params[:email])
     if owner
-      @rubygem.ownerships.create(user: owner)
+      @rubygem.ownerships.create!(user: owner)
```

Comment:
Can we please add test cases for these? ie user should not see `Owner added successfully` if create fails.

Reply:
I've expanded on this and added a check for making sure the owner doesn't already exist. If that the case, the controller will now respond with "Owner already exists.' 
I've also added the test case for raising the exception based on `create!`, but it feels to me like I am testing the implementation details of AR itself. What do you think?

Reply:
I am not sure if we need `create!` here. I am guessing this would raise server error. Just checking the return value of `create` and show `errors.full_messages` should work (something [like this](https://github.com/rubygems/rubygems.org/blob/master/app/controllers/api/v1/deletions_controller.rb#L9)).

`!owner.rubygems.include?(@rubygem)` is covered in Ownership validation and showing `errors` would cover this.

>  I am testing the implementation details of AR itself.

We need to test the branching here(not if AR raises exception). Test need would be something like following:
```
context "when adding owner fails" do
  should "show the error message" do
    assert "owner could not be added...", @response.body
     ...
  end
end
```

Reply:
> I am not sure if we need create! here. I am guessing this would raise server error.

I will do, this get inconsistencies on API level where is expected to respond something meaningful like the action you already mention or [this other fix ](https://github.com/rubygems/rubygems.org/pull/2414/files#r442758239).